### PR TITLE
Add support for <sub> and <sup> in "extra" hover label

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/3357 

cc @plotly/plotly_js and @Braintelligence

----

Note that this PR doesn't fix the alignment problem brought in https://github.com/plotly/plotly.js/issues/3175#issuecomment-448565062 and suspected by @alexcjohnson in https://github.com/plotly/plotly.js/issues/3357#issuecomment-454081020. This issue is now in https://github.com/plotly/plotly.js/issues/3442.

### CODE REVIEW

**Comment by Braintelligence** [comment]
Thank you very much!

File: src/components/fx/hover.js [hunk 1]
```diff
@@ -925,15 +925,10 @@ function createHoverText(hoverData, opts, gd) {
...
-
-            var nameLength = Math.round(d.nameLength);
-
-            if(nameLength > -1 && name.length > nameLength) {
-                if(nameLength > 3) name = name.substr(0, nameLength - 3) + '...';
-                else name = name.substr(0, nameLength);
-            }
+            name = svgTextUtils.plainText(d.name || '', {
+                len: d.nameLength,
+                allowedTags: ['br', 'sub', 'sup']
```
Comment (etpinard) [inline src/components/fx/hover.js]: From our list of pseudo-html tags:

https://github.com/plotly/plotly.js/blob/cc4597257b411688b2bbc6dbe90f8a6e78cc96c8/src/lib/svg_text_utils.js#L233-L244

I can us potentially adding `<b>`, `<i>` and `<em>`. Thoughts?

Comment (archmoj) [reply src/components/fx/hover.js]: In `webGL` texts now we do support `<b>` and `<i>`. So it might be nice to have them here.

Comment (etpinard) [reply src/components/fx/hover.js]: Ok! Done in -> https://github.com/plotly/plotly.js/pull/3443/commits/6c7b0ede2e860ca9b7e03d4a849a4dedbccf2770

**Comment by antoinerg** [comment]
Thanks @etpinard for the improvements :muscle: 

Here's a Codepen showing the fix: https://codepen.io/anon/pen/BMaNox :ok_hand: 

It's a :dancer: for me!

**Pull request merged**